packages/framework/src/Component/ClassExtension/PropertyAnnotationsFactory.php

```diff
@@ -0,0 +1,111 @@
+     * @param \Roave\BetterReflection\Reflection\ReflectionClass $projectClassBetterReflection
+     * @return string
+     */
+    public function getProjectClassNecessaryPropertyAnnotationsLines(
+        ReflectionClass $frameworkClassBetterReflection,
+        ReflectionClass $projectClassBetterReflection
+    ): string {
+        $projectClassDocBlock = $projectClassBetterReflection->getDocComment();
+        $propertyAnnotationsLines = '';
+        foreach ($frameworkClassBetterReflection->getProperties() as $property) {
+            $propertyAnnotationLine = $this->getPropertyAnnotationLine($property, $projectClassBetterReflection);
+            if ($propertyAnnotationLine !== '' && strpos($projectClassDocBlock, $propertyAnnotationLine) === false) {
+                $propertyAnnotationsLines .= $propertyAnnotationLine;
+            }
+        }
+
+        return $propertyAnnotationsLines;
+    }
+
+    /**
+     * @param \Roave\BetterReflection\Reflection\ReflectionProperty $reflectionPropertyFromFrameworkClass
+     * @param \Roave\BetterReflection\Reflection\ReflectionClass $projectClassBetterReflection
+     * @return string
+     */
+    protected function getPropertyAnnotationLine(
+        ReflectionProperty $reflectionPropertyFromFrameworkClass,
+        ReflectionClass $projectClassBetterReflection
+    ): string {
+        foreach ($this->annotationsReplacementsMap->getPatterns() as $frameworkClassPattern) {
+            if (!$this->isPropertyDeclaredInClass($reflectionPropertyFromFrameworkClass->getName(), $projectClassBetterReflection)
+                && $this->isPropertyOfTypeThatIsExtendedInProject($reflectionPropertyFromFrameworkClass, $frameworkClassPattern)
+            ) {
+                $replacedTypeForProperty = $this->annotationsReplacer->replaceInPropertyType($reflectionPropertyFromFrameworkClass);
+
+                return sprintf(
+                    " * @property %s%s $%s\n",
+                    $reflectionPropertyFromFrameworkClass->isStatic() ? 'static ' : '',
+                    $replacedTypeForProperty,
+                    $reflectionPropertyFromFrameworkClass->getName()
+                );
+            }
+        }
+
+        return '';
+    }
+
+    /**
+     * @param string $propertyName
+     * @param \Roave\BetterReflection\Reflection\ReflectionClass $reflectionClass
+     * @return bool
+     */
+    protected function isPropertyDeclaredInClass(string $propertyName, ReflectionClass $reflectionClass)
+    {
+        $reflectionProperty = $reflectionClass->getProperty($propertyName);
+        if ($reflectionProperty === null) {
+            return false;
+        }
+
+        return $reflectionProperty->getDeclaringClass()->getName() === $reflectionClass->getName();
+    }
+
+    /**
+     * @param \Roave\BetterReflection\Reflection\ReflectionProperty $reflectionProperty
+     * @param string $frameworkClassPattern
+     * @return false|int
```

Comment:
false|int doesn't seem right when the method is named isXXXX

Reply:
fixed in a20ac86d0